File: YemenSchoolsV1.Clint/src/app/shared/models/daily-log/daily-log.ts
export interface CreateDailyLogDto {
    lessonCovered: string;       // الدرس الذي تم شرحه
    homeworkAssigned: string;    // الواجب المنزلي
    teacherNotes?: string;       // ملاحظات المعلم (اختياري)
    sectionSubjectId: string;    // معرف الشعبة + المادة
}
export interface DailyLogDto {
    id: string;
    lessonCovered: string;
    homeworkAssigned: string;
    teacherNotes?: string;
    date: string;
    subjectName: string; // أضفنا اسم المادة

    sectionSubjectId: string;
    teacherId: string;
}

export interface DailyLog {
    id: string;
    lessonCovered: string;
    homeworkAssigned: string;
    teacherNotes?: string;
    date: string; // يمكن تحويلها إلى Date في TS
    sectionSubjectId: string;
}

File: YemenSchoolsV1.Domain/Entities/DailyLog.cs
namespace YemenSchoolsV1.Domain.Entities
{
    public class DailyLog
    {
        public Guid Id { get; set; }
        public string? LessonCovered { get; set; }
        public string? HomeworkAssigned { get; set; }
        public string? TeacherNotes { get; set; }
        public DateTime Date { get; set; }

        // المفاتيح الخارجية للربط مع الكيانات الأخرى
        public Guid SectionSubjectId { get; set; }
        public Guid TeacherId { get; set; }

        // خصائص التنقل (Navigation Properties)
        public SectionSubject SectionSubject { get; set; } = null!;
        public Teacher Teacher { get; set; } = null!;
    }
}
